controller/engine_controller.go

```diff
@@ -973,11 +977,31 @@ func (m *EngineMonitor) refresh(engine *longhorn.Engine) error {
 			m.logger.Debug("Cannot restore the backup for engine since it is still in the backoff window")
 			return nil
 		}
+
+		if isReachedLimit, err := m.isReachedConcurrentVolumeBackupRestoreLimit(); err != nil {
+			m.logger.WithError(err).Warn("Failed to check concurrent volume backup restore limit")
+			return nil
+
+		} else if isReachedLimit {
+			m.logger.Debugf("Reached the concurrent volume backup restore limit of %v, retry later", types.SettingNameConcurrentBackupRestorePerNodeLimit)
+			m.restoreBackoff.Next(engine.Name, time.Now())
+			return nil
+		}
```

Comment:
This part should be checked only if the volume is not DR, right?

Should move into the below `if !isDRVolume {`